candy-machine/program/src/lib.rs

```diff
@@ -191,15 +201,17 @@ pub mod candy_machine {
                                 if ctx.accounts.payer.key() != candy_machine.authority
                                     && !ws.presale
                                 {
-                                    return Err(ErrorCode::CandyMachineNotLive.into());
+                                    punish_bots(ErrorCode::CandyMachineNotLive, &payer, &ctx.accounts.token_metadata_program, BOT_FEE)?;
```

Comment:
crush 'em all

Reply:
This is not a solution. All it takes is the frontend dev not knowing how dates work and allowing users to send mint instructions.

Reply:
whip the frontend dev because they didn't make good html then